exchanges/order/order_test.go

```diff
@@ -310,42 +309,22 @@ func TestInferCostsAndTimes(t *testing.T) {
 	t.Parallel()
 	var detail Detail
 	detail.InferCostsAndTimes()
-	if detail.Amount != detail.ExecutedAmount+detail.RemainingAmount {
-		t.Errorf(
-			"Order detail amounts not equals. Expected 0, received %f",
-			detail.Amount-(detail.ExecutedAmount+detail.RemainingAmount),
-		)
-	}
+	assert.Equal(t, detail.Amount, detail.ExecutedAmount+detail.RemainingAmount)
```

Comment:
🐒 
* Need to say what the test is
* Am I being stupid, or are we not inferring that it's zero? In which case, we should be testing that
```suggestion
	assert.Zero(t, detail.Amount, "InferCostsAndTimes on empty details should set correct Amount")
```
( And the same for ExecutedAmount and Remaining amount )